testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -13519,6 +13519,32 @@ void testTimestampDiff(boolean coercionEnabled) {
     f.checkNull("date_add(CAST(NULL AS DATE), interval 5 day)");
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6396">[CALCITE-6396]
+   * Add ADD_MONTHS function (enabled in Spark library)</a>.
+   */
+  @Test void testAddMonths() {
+    final SqlOperatorFixture f0 = fixture()
+        .setFor(SqlLibraryOperators.ADD_MONTHS);
+    f0.checkFails("^add_months(date '2008-12-25', "
+            + "5)^",
+        "No match found for function signature "
+            + "ADD_MONTHS\\(<DATE>, <NUMERIC>\\)", false);
+    final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.SPARK);
+    f.checkScalar("add_months(date '2016-02-22', 2)",
```

Comment:
how about a few tests with other data types than integers?
strings, longs, decimals

Reply:
This will fail during parameter parsing. Calcite currently does not support string, double, and other types for date function calculations.

Reply:
if that's the correct behavior you should leave these tests in.
otherwise, it means the implementation is not complete

Reply:
I read the official document of DataRisks, and the description in the document is only int type
<img width="736" alt="image" src="https://github.com/user-attachments/assets/3a725fbb-459d-4734-adb8-d1437730fa4f">
https://docs.databricks.com/en/sql/language-manual/functions/add_months.html